src/containerapp/azext_containerapp/tests/latest/test_containerapp_up.py

```diff
@@ -37,3 +38,65 @@ def test_containerapp_up_image_e2e(self, resource_group):
         self.assertTrue(resp.ok)
 
         self.cmd(f"containerapp up --image {image} --environment {env_name} -g {resource_group} -n {app_name} -l {TEST_LOCATION.upper()}")
+
+    @ResourceGroupPreparer(location="eastus2")
+    def test_containerapp_up_source_with_buildpack_e2e(self, resource_group):
+        self.cmd('configure --defaults location={}'.format(TEST_LOCATION))
+
+        source_path = os.path.join(TEST_DIR, os.path.join("data", "source_built_using_buildpack"))
```

Comment:
Please consider creating a helper function to share some of the logic between the test cases, as it looks like a good part is duplicated.